AstarteDeviceSDKCSharp/Transport/MQTT/AstarteMqttV1Transport.cs

```diff
@@ -207,5 +216,72 @@ public override async Task ResendAllProperties()
             }
         }
 
+        public override void StartResenderTask()
+        {
+
+            CancellationTokenSource _resenderCancellationTokenSource = new CancellationTokenSource();
+            var cancellationToken = _resenderCancellationTokenSource.Token;
+
+            Task.Run(async () =>
+           {
+               if (!_resendingInProgress)
+               {
+                   await ResendFailedMessages(cancellationToken);
+               }
+           });
+        }
+
+        private async Task ResendFailedMessages(CancellationToken cancellationToken = default)
+        {
+            Trace.WriteLine("Resending stored messages in progress.");
+            _resendingInProgress = true;
+
+            try
+            {
+                if (_client == null || _failedMessageStorage == null)
+                {
+                    Trace.WriteLine("Client or failed message storage is null.");
+                    return;
+                }
+
+                await WaitForPendingMessagesToClear(_client);
+
+                var storedMessages = await _failedMessageStorage.LoadQueuedMessagesAsync();
+                if (storedMessages.Count == 0)
+                {
+                    Trace.WriteLine("No more stored messages to resend.");
+                    return;
```

Comment:
I think before return you have to set `_resendingInProgress` to false.
```suggestion
                    _resendingInProgress = false;
                    return;
```

Reply:
We don't have to because of `finally` block but I did it now.